src/lib/server/websearch/sentenceSimilarity.ts

```diff
@@ -0,0 +1,39 @@
+import type { Tensor } from "@xenova/transformers";
+import { pipeline, dot } from "@xenova/transformers";
+
+// see here: https://github.com/nmslib/hnswlib/blob/359b2ba87358224963986f709e593d799064ace6/README.md?plain=1#L34
+function innerProduct(tensor1: Tensor, tensor2: Tensor) {
+	return 1.0 - dot(tensor1.data, tensor2.data);
+}
+
+const extractor = await pipeline("feature-extraction", "Xenova/e5-small-v2");
```

Comment:
I can make the embedding model env variable. However, there is a specific preprocessing step https://github.com/huggingface/chat-ui/blob/a47c6f29ae752e1e2727bfb367aa98559c6cb1d2/src/lib/server/websearch/sentenceSimilarity.ts#L16-L18

Reply:
really cool cc @xenova @osanseviero ^

Reply:
Haha yes indeed! 🔥 

PS @mishig25, you can experiment with using the unquantized versions of the models too by passing `{quantized: false}` as a third parameter to the pipeline creation function.

i.e., 
```js
const extractor = await pipeline("feature-extraction", "Xenova/e5-small-v2", { quantized: false });
```

The unquantized models are ~4x larger, however, so just take note of that.


> using [Xenova/e5-small-v2](https://huggingface.co/Xenova/e5-small-v2) which provided better results than the popular [Xenova/all-MiniLM-L6-v2](https://huggingface.co/Xenova/all-MiniLM-L6-v2)

The [MTEB leaderboard](https://huggingface.co/spaces/mteb/leaderboard) corroborates this observation.